libs/platform/ebpf_platform.h

```diff
@@ -415,6 +418,41 @@ extern "C"
     bool
     ebpf_queue_non_preemptible_work_item(_In_ ebpf_non_preemptible_work_item_t* work_item, _In_opt_ void* parameter_1);
 
+    /**
+     * @brief Query the platform to determine if preemptible work items are
+     *   supported.
+     *
+     * @retval true Preemptible work items are supported.
+     * @retval false Preemptible work items are not supported.
+     */
+    bool
+    ebpf_is_preemptible_work_item_supported();
+
+    /**
+     * @brief Create a preemptible work item.
+     *
+     * @param[out] work_item Pointer to memory that will contain the pointer to
+     *  the preemptible work item on success.
+     * @param[in] work_item_routine Routine to execute as a work item.
+     * @param[in] work_item_context Context to pass to the routine.
+     * @retval EBPF_SUCCESS The operation was successful.
+     * @retval EBPF_NO_MEMORY Unable to allocate resources for this
+     *  work item.
+     */
+    ebpf_result_t
+    ebpf_allocate_preemptible_work_item(
+        _Outptr_ ebpf_preemptible_work_item_t** work_item,
+        _In_ void (*work_item_routine)(_In_ const void* work_item_context),
```

Comment:
Per your comment about the context being In_opt in IO_WORKITEM_ROUTINE, should this line be:
```suggestion
        _In_ void (*work_item_routine)(_In_opt_ const void* work_item_context),
```
?